plugin/src/test/java/org/opensearch/ml/rest/RestMLRemoteInferenceIT.java

```diff
@@ -0,0 +1,675 @@
+            );
+        Map responseMap = parseResponseToMap(response);
+        assertEquals((String) responseMap.get("status"), "CREATED");
+        String modelGroupId = (String) responseMap.get("model_group_id");
+
+        String registerModelEntity = "{\n"
+            + "  \"name\": \""
+            + name
+            + "\",\n"
+            + "  \"function_name\": \"remote\",\n"
+            + "  \"model_group_id\": \""
+            + modelGroupId
+            + "\",\n"
+            + "  \"version\": \"1.0.0\",\n"
+            + "  \"description\": \"test model\",\n"
+            + "  \"connector_id\": \""
+            + connectorId
+            + "\"\n"
+            + "}";
+        return TestHelper
+            .makeRequest(client(), "POST", "/_plugins/_ml/models/_register", null, TestHelper.toHttpEntity(registerModelEntity), null);
+    }
+
+    private Response deployRemoteModel(String modelId) throws IOException {
+        return TestHelper.makeRequest(client(), "POST", "/_plugins/_ml/models/" + modelId + "/_deploy", null, "", null);
+    }
+
+    private Response predictRemoteModel(String modelId, String input) throws IOException {
+        return TestHelper.makeRequest(client(), "POST", "/_plugins/_ml/models/" + modelId + "/_predict", null, input, null);
+    }
+
+    private Response undeployRemoteModel(String modelId) throws IOException {
+        String undeployEntity = "{\n"
+            + "  \"SYqCMdsFTumUwoHZcsgiUg\": {\n"
+            + "    \"stats\": {\n"
+            + "      \""
+            + modelId
+            + "\": \"undeployed\"\n"
+            + "    }\n"
+            + "  }\n"
+            + "}";
+        return TestHelper.makeRequest(client(), "POST", "/_plugins/_ml/models/" + modelId + "/_undeploy", null, undeployEntity, null);
+    }
+
+    private Map parseResponseToMap(Response response) throws IOException {
+        HttpEntity entity = response.getEntity();
+        assertNotNull(response);
+        String entityString = TestHelper.httpEntityToString(entity);
+        return gson.fromJson(entityString, Map.class);
+    }
+
+    private void disableClusterConnectorAccessControl() throws IOException {
+        Response response = TestHelper
+            .makeRequest(
+                client(),
+                "PUT",
+                "_cluster/settings",
+                null,
+                "{\"persistent\":{\"plugins.ml_commons.connector_access_control_enabled\":false}}",
+                ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, ""))
+            );
+        assertEquals(200, response.getStatusLine().getStatusCode());
+    }
+
+    private void setEncryptionMasterKey() throws IOException {
+        Response response = TestHelper
+            .makeRequest(
+                client(),
+                "PUT",
+                "_cluster/settings",
+                null,
+                "{\"persistent\":{\"plugins.ml_commons.encryption.master_key\":\"0000000000000011\"}}",
```

Comment:
We removed this master key in this PR https://github.com/opensearch-project/ml-commons/pull/1075. So this method will fail. Can you rebase and make sure all CI can pass?